lib/chat_api_web/controllers/registration_controller.ex

```diff
@@ -6,38 +6,67 @@ defmodule ChatApiWeb.RegistrationController do
   alias ChatApiWeb.ErrorHelpers
 
   @spec create(Conn.t(), map()) :: Conn.t()
-  def create(conn, %{"user" => user_params}) do
-    invite_token = user_params["invite_token"]
+  def create(conn, params)
 
-    account =
-      if invite_token do
-        ChatApi.UserInvitations.get_user_invitation!(user_params["invite_token"]).account
-      else
-        {:ok, account} =
-          ChatApi.Accounts.create_account(%{company_name: user_params["company_name"]})
+  def create(conn, %{"user" => user_params}) when is_map_key(user_params, "invite_token") do
+    try do
+      invite = ChatApi.UserInvitations.get_user_invitation!(user_params["invite_token"])
+      params = Enum.into(user_params, %{"account_id" => invite.account.id})
+      conn
+      |> Pow.Plug.create_user(params)
+      |> case do
+           {:ok, _user, conn} ->
+             if ChatApi.UserInvitations.expired?(invite) do
+               send_server_error(conn, "Invitation token has expired")
```

Comment:
hmm I wonder if we should use a 403 status for these errors instead of 500... not blocking for this PR, but just a note for the future

Reply:
Sure thing! That's a more informative error code